src/qt/statusbar.cpp

```diff
@@ -102,6 +102,11 @@ bool wxStatusBar::GetFieldRect(int i, wxRect& rect) const
     wxCHECK_MSG( (i >= 0) && ((size_t)i < m_panes.GetCount()), false,
                  "invalid statusbar field index" );
 
+    const_cast<wxStatusBar *>(this)->UpdateFields();
+
+    if (m_qtPanes.empty())
+        return false;
+
```

Comment:
I don't mind keeping this for safety reasons. But, is the crash reproducible in the **statbar** sample ?

Reply:
Here's the usage in KiCad: https://gitlab.com/kicad/code/kicad/-/blob/07d7ccf46fbbaf3c81a56deb8e2e4302ccdbcd47/common/widgets/kistatusbar.cpp#L225

`GetFieldRect` is called before `SetStatusText`, so `m_qtPanes` is empty (if `UpdateFields` isn't called).
"statbar" sample doesn't seem to do that.

if `UpdateFields` _is_ called, `m_qtPanes` will probably never be empty.

Reply:
I see, thanks. so we can rename `UpdateFields()` to `CreateFieldsIfNeeded()` and call it from this function as well as from `DoUpdateStatusText()` and remove this test (i.e. `if ( m_qtPanes.empty() )` ) because the check at the beginning of the function is enough. IOW, if  `m_panes` is empty, the function will assert.

TIA.